etc/systemd/system-generators/zfs-mount-generator.in

```diff
@@ -46,6 +46,8 @@ fi
 # avoid regressions, this dependency is reduced to "wants" rather than
 # "requires". **THIS MAY CHANGE**
 req_dir="${dest_norm}/local-fs.target.wants/"
+zvol_reqdir="${dest_norm}/zfs-volumes.target.requires"
```

Comment:
Now that we integrated https://github.com/zfsonlinux/zfs/pull/8975, do you think we still need to add an additional `requires` on `zfs-volumes.target`?

Reply:
I believe so (enabling the `zfs-volume@` units should presumably allow the user to disable `zfs-volume-wait.service`).

I feel like I may be misunderstanding something.

Reply:
My original idea was to only use `zfs-volume-wait.service` as a dependency for `zfs-volumes.target`, but that's mostly because using it along with `zfs-volume@` as a dependency was redundant. Disabling `zfs-volume-wait.service` when we are using `zfs-volume@` would be another way to do this.